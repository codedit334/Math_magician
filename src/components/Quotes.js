import React, { useEffect, useState } from 'react';
import Book from '../assets/Book.gif';
import '../styles/quotes.css';

export default function Quotes() {
  const [quote, setQuote] = useState('');
  const [spinner, setSpinner] = useState(false);
  const [displayError, setError] = useState('');

  let spinnerStyle;

  useEffect(() => {
    setError('');
  }, [quote]);

  useEffect(() => {
    const fetchQuote = async () => {
      setSpinner(true);
      try {
        const URL = 'https://api.api-ninjas.com/v1/quotes';
        const res = await fetch(URL, {
          headers: {
            'X-Api-Key': '9gIDMqe2KdUO/d++zPtwQw==fdgQmNP1uakeUI1d',
          },
        });
        const json = await res.json();
        setSpinner(false);
        setQuote(`"${json[0].quote}" by ${json[0].author}`);
      } catch (error) {
        const err = `Error: ${error}`;
        setSpinner(false);
        setError(err);
      }
    };
    fetchQuote();
  }, []);

  if (!spinner) { spinnerStyle = { display: 'none' }; }
  return (
    <div className="display-quote">
      <img src={Book} alt="loading..." style={spinnerStyle} />
      <div className="quote">
        {quote}
      </div>
      <span className="error">{displayError}</span>
    </div>
  );
}
