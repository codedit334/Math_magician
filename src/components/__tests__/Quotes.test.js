import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Quotes from '../Quotes';

test('Sould render Quotes Component', () => {
  render(<Quotes />);
  const QuoteElement = screen.getByTestId('Quotes');
  expect(QuoteElement).toBeInTheDocument();

  const tree = renderer.create(<Quotes />).toJSON();
  expect(tree).toMatchSnapshot();
});
