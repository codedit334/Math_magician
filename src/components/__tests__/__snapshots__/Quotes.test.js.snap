// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sould render CalcKeys Component 1`] = `
<div
  className="display-quote"
  data-testid="Quotes"
>
  <img
    alt="loading..."
    src="Book.gif"
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    className="quote"
  />
  <span
    className="error"
  />
</div>
`;

exports[`Sould render Quotes Component 1`] = `
<div
  className="display-quote"
  data-testid="Quotes"
>
  <img
    alt="loading..."
    src="Book.gif"
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    className="quote"
  />
  <span
    className="error"
  />
</div>
`;
